src/components/NewListing/NewListingForm/NewListingForm.tsx

```diff
@@ -29,13 +27,55 @@ const NewListingForm = ({ onPublish }: Props) => {
   const [isAccidentFree, setIsAccidentFree] = useState(false);
   const [images, setImages] = useState<File[] | []>([]);
 
+  const [dropdownData, setDropdownData] = useState<dropdownData | null>(null);
+
   const { user } = useAuth() as AuthType;
 
+  useEffect(() => {
+    const fetchDropdownData = async () =>
+      setDropdownData(await getDropdownData());
+    fetchDropdownData();
+  }, []);
+
+  const setImagesHandler = (uploadedImages: File[] | []) =>
+    setImages(uploadedImages);
+
+  const numInputsHandler = (
+    event: ChangeEvent<HTMLInputElement>,
+    input: string
+  ) => {
+    const value = event.target.value.replace(/\D/g, '');
+
+    if (value === '0') return;
+
+    if (input === 'power') setPower(value);
```

Comment:
tutaj bardziej pasowałby switch no ale to is how is

Reply:
słuszna uwaga